### PR TITLE
refactor request.js

### PR DESCRIPTION
**Pull request by enguerran**
I also added tests:

```
 PASS  packages/cozy-konnector-libs/src/libs/request/request.spec.js
  requestFactory
    get request options
      ✓ without cheerio (3ms)
      ✓ with cheerio (1ms)
    transform with cheerio
      ✓ get cheerio object (245ms)
      ✓ accept resolveWithFullResponse option (1ms)
    merge default options
      ✓ should get default options (1ms)
      ✓ should update default options with given ones
      ✓ should deactivate json with cheerio on
    when loading a `latin1` encoded page
      ○ skipped 2 tests

------------|----------|----------|----------|----------|-------------------|
File        |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------|----------|----------|----------|----------|-------------------|
 request.js |    86.67 |    77.78 |       80 |    86.67 |             59,60 |
------------|----------|----------|----------|----------|-------------------|
```

### CODE REVIEW

File: packages/cozy-konnector-libs/src/libs/request/request.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+function getRequestOptions({ cheerio, userAgent, ...options }) {
+  return cheerio
+    ? {
+        ...options,
+        transform: transformWithCheerio,
+        headers: {
+          ...options.headers,
+          'User-Agent': getUserAgent(true)
```
Comment (aenario) [inline packages/cozy-konnector-libs/src/libs/request/request.js]: If `options.headers` has an `User-Agent`, is should take priority

File: packages/cozy-konnector-libs/src/libs/request/request.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        }
+      }
+    : {
+        ...options,
+        headers: {
+          ...options.headers,
+          'User-Agent': getUserAgent(userAgent)
```
Comment (aenario) [inline packages/cozy-konnector-libs/src/libs/request/request.js]: idem : If options.headers has an User-Agent, is should take priority

**Comment by doubleface** [comment]
Thanks for this much needed refacto @enguerran :+1:

**Comment by enguerran** [comment]
@aenario I did fixup previous commits to be more consistent with previous behavior regarding the User-Agent, I adapted the test file to cover as many combinations as possible.

**Pull request merged**